autopush/router/gcm.py

```diff
@@ -84,7 +85,13 @@ def _route(self, notification, router_data):
                 data['enckey'] = notification.headers.get('encryption-key')
             if ('crypto-key' in notification.headers):
                 data['cryptokey'] = notification.headers.get('crypto-key')
-            data['body'] = notification.data
+            udata = urlsafe_b64encode(notification.data)
+            mdata = self.config.get('max_data', 4096)
+            if len(udata) > mdata:
+                self._error("Converted buffer too long by %d bytes" % (
```

Comment:
I kind of wish I had mentioned earlier that self._error should return the exception, and the raise should go here, just so the code flow is more obvious. As is, it's not immediately apparent that self._error is intentionally going to break the flow here by raising an exception.